llvm/lib/IR/Verifier.cpp

```diff
@@ -5235,6 +5235,25 @@ void Verifier::visitIntrinsicCall(Intrinsic::ID ID, CallBase &Call) {
     }
     break;
   }
+  case Intrinsic::ucmp:
+  case Intrinsic::scmp: {
+
+    Type *SrcTy = Call.getOperand(0)->getType();
+    Type *DestTy = Call.getType();
+
+    Check(SrcTy->isIntOrIntVectorTy(), "[u]scmp only operates on integers", Call);
+    Check(DestTy->isIntOrIntVectorTy(), "[u]scmp only produces integers", Call);
+    Check(DestTy->getScalarSizeInBits() >= 2, "DestTy must be at least 2 bits wide", Call);
```

Comment:
Could you please also add a test in llvm/test/Verifier to check this condition, and the one below?

Reply:
Sure thing!

Reply:
I've added a `llvm/test/Verifier/intrinsic-cmp.ll` test file where I'll add a test for the 
* ` Check(DestTy->getScalarSizeInBits() >= 2, ...)` and
* `Check(SrcTy->isVectorTy() == isDestTypeVector, ...)`

Both in that same file, does that seem like the right approach here?

It's my first time setting up the `RUN; ...` unit tests and I'm trying to make sure I'm not missing something obvious here.

Reply:
I don't see the file, maybe you forgot to git add it?

But generally, yes, for verifier tests you can check multiple error conditions in one file.

Reply:
Yes, I haven't pushed it yet. Will try to do it later today.

Reply:
Pushed a change.